module7/Homework7-start/Homework7/Models/UserData.swift

```diff
@@ -0,0 +1,195 @@
+/// distribute, sublicense, create a derivative work, and/or sell copies of the
+/// Software in any work that is designed, intended, or marketed for pedagogical or
+/// instructional purposes related to programming, coding, application development,
+/// or information technology.  Permission for such use, copying, modification,
+/// merger, publication, distribution, sublicensing, creation of derivative works,
+/// or sale is expressly withheld.
+/// 
+/// This project and source code may use libraries or frameworks that are
+/// released under various Open-Source licenses. Use of those libraries and
+/// frameworks are governed by their own individual licenses.
+///
+/// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+/// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+/// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+/// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+/// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+/// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+/// THE SOFTWARE.
+
+import Foundation
+
+struct UserData: Codable, Equatable {
+  let name: Name?
+  let gender: String
+  let location: Location
+  let email: String
+  let login: Login
+  let dob: DOB
+  let registered: Registered
+  let phone: String
+  let cell: String
+  let id: UserID
+  let picture: Picture
+  let nat: String
+}
+
+struct Name: Codable, Equatable {
+  let title: String
+  let first: String
+  let last: String
+
+  var fullName: String {
+    title + " " + first + " " + last
+  }
+}
+
+struct Login: Codable, Equatable {
+  let uuid: String
+  let username: String
+  let password: String
+  let salt: String
+  let md5: String
+  let sha1: String
+  let sha256: String
+}
+
+struct DOB: Codable, Equatable {
+  let date: String
+  let age: Int
+}
+
+struct Registered: Codable, Equatable {
+  let date: String
+  let age: Int
+}
+
+struct UserID: Codable, Equatable {
+  let name: String
+  let value: String
+}
+
+struct Picture: Codable, Equatable {
+  let large: String
+  let medium: String
+  let thumbnail: String
+}
+
+struct Location: Codable, Equatable {
+  let street: Street?
+  let city: String?
+  let state: String?
+  let country: String?
+  let postcode: String?
+  let coordinates: LatLng?
+  let timezone: Timezone?
+
+  enum CodingKeys: String, CodingKey {
+    case street, city, state, country, postcode, coordinates, timezone
+  }
+
+  init(from decoder: Decoder) throws {
+    let container = try decoder.container(keyedBy: CodingKeys.self)
+
+    street = try? container.decode(Street.self, forKey: .street)
+    city = try? container.decode(String.self, forKey: .city)
+    state = try? container.decode(String.self, forKey: .state)
+    country = try? container.decode(String.self, forKey: .country)
+
+    do {
```

Comment:
I was thinking about this more and what you have here solves the problem. Another thing you could do would be to make a `let` of type `Any` and decode to that and then just do `String(<yourVar>)` and let Swift decide how to do it, or to fail and give you a nil value.